unstructured/ingest/connector/salesforce.py

```diff
@@ -0,0 +1,328 @@
+        case = case_template.substitute(
+            id=case_json.get("Id"),
+            type=case_json.get("Type"),
+            status=case_json.get("Status"),
+            reason=case_json.get("Reason"),
+            origin=case_json.get("Origin"),
+            subject=case_json.get("Subject"),
+            priority=case_json.get("Priority"),
+            description=case_json.get("Description"),
+            comments=case_json.get("Comments"),
+        )
+        return dedent(case)
+
+    def create_campaign(self, campaign_json: Dict[Any, Any]) -> str:
+        """Creates partitionable campaign file"""
+        campaign = campaign_template.substitute(
+            id=campaign_json.get("Id"),
+            name=campaign_json.get("Name"),
+            type=campaign_json.get("Type"),
+            status=campaign_json.get("Status"),
+            start_date=campaign_json.get("StartDate"),
+            end_date=campaign_json.get("EndDate"),
+            budgeted_cost=campaign_json.get("BudgetedCost"),
+            actual_cost=campaign_json.get("ActualCost"),
+            description=campaign_json.get("Description"),
+            number_of_leads=campaign_json.get("NumberOfLeads"),
+            number_of_converted_leads=campaign_json.get("NumberOfConvertedLeads"),
+        )
+        return dedent(campaign)
+
+    def create_eml(self, email_json: Dict[Any, Any]) -> str:
+        """Recreates standard expected .eml format using template."""
+        eml = email_template.substitute(
+            date=formatdate(parser.parse(email_json.get("MessageDate")).timestamp()),
+            message_identifier=email_json.get("MessageIdentifier"),
+            subject=email_json.get("Subject"),
+            from_email=email_json.get("FromAddress"),
+            to_email=email_json.get("ToAddress"),
+            textbody=email_json.get("TextBody"),
+        )
+        return dedent(eml)
+
+    @BaseIngestDoc.skip_if_file_exists
+    @requires_dependencies(["simple_salesforce"], extras="salesforce")
+    def get_file(self):
+        """Saves individual json records locally."""
+        self._create_full_tmp_dir_path()
+
+        logger.debug(f"Writing file {self.record.get('Id')} - PID: {os.getpid()}")
+
+        try:
+            if self.record_type == "EmailMessage":
+                formatted_record = self.create_eml(self.record)
+            elif self.record_type == "Account":
+                formatted_record = self.create_account(self.record)
+            elif self.record_type == "Lead":
+                formatted_record = self.create_lead(self.record)
+            elif self.record_type == "Case":
+                formatted_record = self.create_case(self.record)
+            elif self.record_type == "Campaign":
+                formatted_record = self.create_campaign(self.record)
+            else:
+                raise MissingCategoryError(
+                    f"There are no categories with the name: {self.record_type}",
+                )
```

Comment:
can we filter out the missing categories back in the SalesforceConnector `get_ingest_docs` call?

Reply:
Yep. Good idea.